src/components/DayPickerInstrumented.jsx

```diff
@@ -0,0 +1,365 @@
+} from '../../constants';
+
+const propTypes = DateRangePickerShape;
+
+const defaultProps = {
+  startDateId: START_DATE,
+  endDateId: END_DATE,
+  focusedInput: null,
+  minimumNights: 1,
+  isDayBlocked: () => false,
+  disabledDays: [],
+  isOutsideRange: day => !isInclusivelyAfterDay(day, moment()),
+  enableOutsideDays: false,
+  numberOfMonths: 2,
+  showClearDates: false,
+  disabled: false,
+  reopenPickerOnClearDates: false,
+  initialVisibleMonth: () => moment(),
+
+  orientation: HORIZONTAL_ORIENTATION,
+  withPortal: false,
+  withFullScreenPortal: false,
+
+  onDatesChange() {},
+  onFocusChange() {},
+  onPrevMonthClick() {},
+  onNextMonthClick() {},
+
+  // i18n
+  displayFormat: () => moment.localeData().longDateFormat('L'),
+  monthFormat: 'MMMM YYYY',
+  phrases: {
+    closeDatePicker: 'Close',
+    clearDates: 'Clear Dates',
+  },
+};
+
+export default class DayPickerInstrumented extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hoverDate: null,
+    };
+
+    this.isTouchDevice = isTouchDevice();
+
+    this.onOutsideClick = this.onOutsideClick.bind(this);
+    this.onDayMouseEnter = this.onDayMouseEnter.bind(this);
+    this.onDayMouseLeave = this.onDayMouseLeave.bind(this);
+    this.onDayClick = this.onDayClick.bind(this);
+
+    this.onClearFocus = this.onClearFocus.bind(this);
+    this.onStartDateChange = this.onStartDateChange.bind(this);
+    this.onStartDateFocus = this.onStartDateFocus.bind(this);
+    this.onEndDateChange = this.onEndDateChange.bind(this);
+    this.onEndDateFocus = this.onEndDateFocus.bind(this);
+    this.clearDates = this.clearDates.bind(this);
+  }
+
+  onClearFocus() {
+    this.props.onFocusChange(null);
+  }
+
+  onDayClick(day, modifiers, e) {
+    const { minimumNights } = this.props;
+    if (e) e.preventDefault();
+    if (includes(modifiers, 'blocked')) return;
+
+    const { focusedInput } = this.props;
+    let { startDate, endDate } = this.props;
+
+    if (focusedInput === START_DATE) {
+      this.props.onFocusChange(END_DATE);
+
+      startDate = day;
+
+      if (isInclusivelyAfterDay(day, endDate)) {
+        endDate = null;
+      }
+    } else if (focusedInput === END_DATE) {
+      const firstAllowedEndDate = startDate && startDate.clone().add(minimumNights, 'days');
+
+      if (!startDate) {
+        endDate = day;
+        this.props.onFocusChange(START_DATE);
+      } else if (isInclusivelyAfterDay(day, firstAllowedEndDate)) {
+        endDate = day;
+        this.props.onFocusChange(null);
+      } else {
+        startDate = day;
+        endDate = null;
+      }
+    }
+
+    this.props.onDatesChange({ startDate, endDate });
+  }
+
+  onDayMouseEnter(day) {
+    if (this.isTouchDevice) return;
+
+    this.setState({
+      hoverDate: day,
+    });
+  }
+
+  onDayMouseLeave() {
+    if (this.isTouchDevice) return;
+
+    this.setState({
+      hoverDate: null,
+    });
+  }
+
+  onEndDateChange(endDateString) {
```

Comment:
I think that `onEndDateChange` and `onStartDateChange` belong in the `DateRangePicker` still. As do the `onFocusChange` handlers. Basically, anything that deals with the inputs should be in `DateRangePicker`, anything that deals with the modifiers/`DayPicker` only should be here.